### PR TITLE
fddev: add flame command

### PR DESCRIPTION
**Pull request by mmcgee-jump**

### CODE REVIEW

File: config/extra/with-x86-64.mk [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CPPFLAGS+=-fomit-frame-pointer -mfpmath=sse -falign-functions=32
```
[the remark this answers is not in the document]
Comment (mmcgee-jump) [reply config/extra/with-x86-64.mk:1]: https://www.brendangregg.com/blog/2024-03-17/the-return-of-the-frame-pointers.html

File: src/app/fdctl/Local.mk [hunk 2]
```diff
@@ -41,10 +41,13 @@ cargo:
 # grained.
 ifeq ($(RUST_PROFILE),release)
 cargo:
-	cd ./solana && env --unset=LDFLAGS ./cargo build --release --lib -p solana-validator -p solana-genesis -p solana-cli --bin solana
+	cd ./solana && env --unset=LDFLAGS RUSTFLAGS="-C target-cpu=native" ./cargo build --release --lib -p solana-validator -p solana-genesis -p solana-cli --bin solana
```
[the remark this answers is not in the document]
Comment (mmcgee-jump) [reply src/app/fdctl/Local.mk]: Ok, moved this out to native.mk. Left the other ones as default for now.

File: src/util/sandbox/fd_sandbox.c [hunk 3]
```diff
@@ -327,3 +327,26 @@ fd_sandbox_getpid( void ) {
 
   return result;
 }
+
+ulong
+fd_sandbox_gettid( void ) {
+  char tid[ 27 ] = {0}; /* 10 characters for INT_MAX, twice, + /task/ and then a NUL terminator. */
+  long count = readlink( "/proc/thread-self", tid, sizeof(tid) );
```
Comment (ptaffet-jump) [inline src/util/sandbox/fd_sandbox.c:334]: `readlink` does not append a nul terminator, so you should probably pass in `sizeof(tid)-1` or set `tid[26]='\0'`.

Comment (ptaffet-jump) [reply src/util/sandbox/fd_sandbox.c:334]: Although I think your computation of 27 is right, so it should always leave a char at the end that is 0.

**Pull request merged**